lib/webpacker/env.rb

```diff
@@ -0,0 +1,28 @@
+# Singleton registry for setting NODE_ENV based on config/webpack/paths.yml
+# Returns NODE_ENV, if present and is defined in config/webpack/paths.yml
+require "webpacker/file_loader"
+
+class Webpacker::Env < Webpacker::FileLoader
+  class << self
+    def current
+      raise Webpacker::FileLoader::FileLoaderError.new("Webpacker::Env.load must be called first") unless instance
+      instance.data
+    end
+
+    def development?
+      current == "development"
+    end
+
+    def file_path
+      Rails.root.join("config", "webpack", "paths.yml")
+    end
+  end
+
+  private
+    def load
+      environments = File.exist?(@path) ? YAML.load(File.read(@path)).keys : [].freeze
+      return ENV["NODE_ENV"] if environments.include?(ENV["NODE_ENV"])
+      return ENV["NODE_ENV"] = Rails.env if environments.include?(Rails.env)
```

Comment:
Do we need to set `ENV["NODE_ENV"]`, are we even using that?

Can't we just have it available as `Webpacker::Env.current` whoever needs that?